stone/target/js_client.py

```diff
@@ -73,8 +85,33 @@ def generate(self, api):
                         api.route_schema, namespace, route, extra_args)
 
             self.emit()
+
             self.emit('module.exports = routes;')
 
+        typedef_path = os.path.join(self.target_folder_path, self.args.typedef_filename)
+        existing_typedefs = ''
+        # Set of typedef names indicating if it should be skipped.
+        skip_typedefs = set()
+
+        if os.path.isfile(typedef_path):
+            with open(typedef_path, 'r') as typedef_file:
+                existing_typedefs = typedef_file.read()
+            # Looking for patterns like so:
+            # @typedef {base type} TypeName
+            for m in re.finditer('@typedef\s*\{[\w+]+\}\s*(\w+)\s*\n', existing_typedefs):
+                skip_typedefs.add(m.group(1))
+
+        with self.output_to_relative_path(self.args.typedef_filename):
+            # Re-emit existing typedefs, ensuring that a newline occurs
+            # at the end.
+            if existing_typedefs[-1:] is not '\n':
```

Comment:
`[-1:]` asks for all characters after and including the last character. Since we know that's only one character, you should just use `[-1]`.

You should use `!=` instead of `is not` which is normally reserved for singletons (`None`). It's less preferred, but still reasonable here to have used `existing_typedefs.endswith('\n')`.